### PR TITLE
cli/cmd: cleanups part 3

### PR DESCRIPTION
**Pull request by invidian**
~**Includes commits from #1015**~

Part of #603.

### CODE REVIEW

File: cli/cmd/component-delete.go [hunk 1]
```diff
@@ -63,14 +63,24 @@ func runDelete(cmd *cobra.Command, args []string) {
 		log.SetLevel(log.DebugLevel)
 	}
 
-	if err := componentDelete(contextLogger, args); err != nil {
+	options := componentDeleteOptions{
+		confirm:         confirm,
+		deleteNamespace: deleteNamespace,
+	}
+
+	if err := componentDelete(contextLogger, args, options); err != nil {
```
Comment (ipochi) [inline cli/cmd/component-delete.go:72]: why not also add args [components to delete] in the options struct ?

Comment (invidian) [reply cli/cmd/component-delete.go:72]: Hm, I imagine they act on sort of different levels. `args` and `contextLogger` as invokation-specific, while options struct could possibly be re-used between multiple `componentDelete()` calls.

**Review by ipochi — DISMISSED** [review]
LGTM :tada:

**Comment by invidian** [comment]
@ipochi I had to rebase, one more LGTM?

**Review by ipochi — DISMISSED** [review]

**Comment by knrt10** [comment]
CI was not triggering, so I had to re-push

File: cli/cmd/utils.go [hunk 2]
```diff
@@ -187,7 +187,10 @@ func readKubeconfigFromTerraformState(contextLogger *log.Entry) ([]byte, error)
 	contextLogger.Warn("Kubeconfig file not found in assets directory, pulling kubeconfig from " +
 		"Terraform state, this might be slow. Run 'lokoctl cluster apply' to fix it.")
 
-	c, err := initialize(contextLogger)
+	// TODO: Add Terraform verbose support back.
```
Comment (iaguis) [inline cli/cmd/utils.go:188]: I understand this means you can't configure verboseness when reading the kubeconfig from the terraform state.

In that case I think I'm fine with this.

File: cli/cmd/cluster-apply.go [hunk 3]
```diff
@@ -111,7 +111,11 @@ func clusterApply(contextLogger *log.Entry, options clusterApplyOptions) error {
 
 	fmt.Printf("\nYour configurations are stored in %s\n", c.assetDir)
 
-	kubeconfig, err := getKubeconfig(contextLogger, c.lokomotiveConfig, true)
+	kg := kubeconfigGetter{
```
Comment (iaguis) [inline cli/cmd/cluster-apply.go:114]: I'm not super convinced by this getter (especially regarding naming) but I think this is an improvement over the current state so I'm not blocking this.

**Review by iaguis — CHANGES_REQUESTED** [review]
Small nit.

File: cli/cmd/component-delete.go [hunk 4]
```diff
@@ -63,16 +63,24 @@ func runDelete(cmd *cobra.Command, args []string) {
 		log.SetLevel(log.DebugLevel)
 	}
 
+	if err := componentDelete(contextLogger, args); err != nil {
+		contextLogger.Fatalf("Deleting components failed: %v", err)
+	}
+}
+
+// componentApply implements 'lokoctl component delete' separated from CLI
```
[the remark this answers is not in the document]
Comment (knrt10) [reply cli/cmd/component-delete.go:83]: I will fix this

[the remark this answers is not in the document]
Comment (knrt10) [reply cli/cmd/component-delete.go]: Done

**Review by surajssd — APPROVED** [review]
looks ok

**Review by iaguis — APPROVED** [review]
lgtm

File: cli/cmd/cluster-apply.go (hunk 3 above)
Comment (invidian) [reply cli/cmd/cluster-apply.go:114]: Yeah, once moved to separate package, we need to continue refactoring...

**Comment by invidian** [comment]
Thanks for merging!

**Pull request merged**